### PR TITLE
Benchmark suite of team 1 and adaption for the code of team 2 thereof

### PR DESCRIPTION
**Pull request by sgraf812**
Benchmark results captured from the reference hardware can be found in the text file at the root of the repo.

This is a more involved change, so I thought first handling and discussing it via a PR might be more appropriate.

### CODE REVIEW

**Comment by dopitz** [comment]
I don't know if we should disable all the warnings when compiling the celero API. 

Where did you get the numbers for the baseline? I'd suggest to take them from one of our earlier builds without optimization, then we wouldn't have to admit, that the baseline is useless and can be ignored.
Either run the benchmark suite on an old version and record the values again, or use a commit somewhere around this one https://semaphoreci.com/sgraf/mlalign/branches/master/builds/26 where the baseline should be probably exactly what we want as far as I remember.

**Comment by nbaudis** [comment]
I think the baseline values are already taken from one of our earlier builds, but I don't know if it was completely unoptimized. Maybe we could use the reference implementation here if a baseline actually is required.

The PR looks fine to me.

**Comment by stamatak** [comment]
It's a bit hard to follow exactly what you did here, so maybe we can 
just discuss this tonight?

alexis

On 29.10.2015 08:02, nbaudis wrote:

> I think the baseline values are already taken from one of our earlier
> builds, but I don't know if it was completely unoptimized. Maybe we
> could use the reference implementation here if a baseline actually is
> required.
> 
> The PR looks fine to me.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/nbaudis/bioinf2015/pull/1#issuecomment-152098658.

## 

Alexandros (Alexis) Stamatakis

Research Group Leader, Heidelberg Institute for Theoretical Studies
Full Professor, Dept. of Informatics, Karlsruhe Institute of Technology
Adjunct Professor, Dept. of Ecology and Evolutionary Biology, University
of Arizona at Tucson

www.exelixis-lab.org

**Comment by sgraf812** [comment]
@Wh0p At one point we already tried to disable all the warnings for the Celero subproject, but I remember that it wasn't so easy to have different flags for a subproject. Worth looking into though, because it severely uglifies the build process. The baseline values were taken from one of our earlier, not fully optimized builds, I think. I'd say we take the values from team 2 as a baseline.

@stamatak Sure, as I said the changes are quite involved. The team 1 impl is actually a fairly straight copy of the repository we worked with, we just stripped out the benchmarking stuff for submission because Celero (the benchmarking library) generates quite a few warnings on the highest warning level. Also it  wasn't strictly needed for our submission, so we omitted it for the sake of project complexity.

**Comment by stamatak** [comment]
Thanks, I see and I agree that the benchmarking stuff is probably not 
needed,

Alexis

On 29.10.2015 10:21, Sebastian Graf wrote:

> @Wh0p https://github.com/Wh0p At one point we already tried to disable
> all the warnings for the Celero subproject, but I remember that it
> wasn't so easy to have different flags for a subproject. Worth looking
> into though, because it severely uglifies the build process. The
> baseline values were taken from one of our earlier, not fully optimized
> builds, I think. I'd say we take the values from team 2 as a baseline.
> 
> @stamatak https://github.com/stamatak Sure, as I said the changes are
> quite involved. The team 1 impl is actually a fairly straight copy of
> the repository we worked with, we just stripped out the benchmarking
> stuff for submission because Celero (the benchmarking library) generates
> quite a few warnings on the highest warning level. Also it wasn't
> strictly needed for our submission, so we omitted it for the sake of
> project complexity.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/nbaudis/bioinf2015/pull/1#issuecomment-152124616.

## 

Alexandros (Alexis) Stamatakis

Research Group Leader, Heidelberg Institute for Theoretical Studies
Full Professor, Dept. of Informatics, Karlsruhe Institute of Technology
Adjunct Professor, Dept. of Ecology and Evolutionary Biology, University
of Arizona at Tucson

www.exelixis-lab.org

**Comment by sgraf812** [comment]
So, I suggest we just take the benchmark results and park this pull request, OK?

**Comment by stamatak** [comment]
fine with me,

alexis

On 30.10.2015 16:49, Sebastian Graf wrote:

> So, I suggest we just take the benchmark results and park this pull
> request, OK?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/nbaudis/bioinf2015/pull/1#issuecomment-152562697.

## 

Alexandros (Alexis) Stamatakis

Research Group Leader, Heidelberg Institute for Theoretical Studies
Full Professor, Dept. of Informatics, Karlsruhe Institute of Technology
Adjunct Professor, Dept. of Ecology and Evolutionary Biology, University
of Arizona at Tucson

www.exelixis-lab.org
